src/main/java/org/wildfly/security/auth/client/ElytronXmlParser.java

```diff
@@ -1419,6 +1442,265 @@ static Module parseModuleRefType(ConfigurationXMLStreamReader reader) throws Con
         throw reader.unexpectedDocumentEnd();
     }
 
+    /**
+     * Parse an XML element of type {@code set-bearer-token} from an XML reader.
+     *
+     * @param reader the XML stream reader
+     * @param bearerTokensMap the map of bearer tokens to use
+     * @return the bearer token entry factory
+     * @throws ConfigXMLParseException if the resource failed to be parsed
+     */
+    static SecurityFactory<CredentialSource> parseBearerTokenRefType(ConfigurationXMLStreamReader reader, final Map<String, SecurityFactory<CredentialSource>> bearerTokensMap) throws ConfigXMLParseException {
+        final int attributeCount = reader.getAttributeCount();
+        String name = null;
+        for (int i = 0; i < attributeCount; i ++) {
+            checkAttributeNamespace(reader, i);
+            switch (reader.getAttributeLocalName(i)) {
+                case "name": {
+                    if (name != null) throw reader.unexpectedAttribute(i);
+                    name = reader.getAttributeValue(i);
+                    break;
+                }
+                default:
+
+            }
+        }
+        if (name == null) {
+            throw missingAttribute(reader, "name");
+        }
+        while (reader.hasNext()) {
+            final int tag = reader.nextTag();
+            if (tag == START_ELEMENT) {
+                throw reader.unexpectedElement();
+            } else if (tag == END_ELEMENT) {
+                String finalName = name;
+                return () -> {
+                    SecurityFactory<CredentialSource> bearerTokenCallbackHandler = bearerTokensMap.get(finalName);
+                    if (bearerTokenCallbackHandler == null) {
+                        throw xmlLog.unknownBearerTokenConfigSpecified(finalName);
+                    }
+                    try {
+                        return bearerTokenCallbackHandler.create();
+                    } catch (GeneralSecurityException e) {
```

Comment:
This is not necessary since the enclosing return type already can throw GSE.  It's probably best to just pass the exception up without rewriting it, unless there's a good reason to wrap it.